File: api/src/main/java/org/pmiops/workbench/db/dao/UserRecentResourceService.java
package org.pmiops.workbench.db.dao;

import java.util.List;
import org.pmiops.workbench.db.model.DbUserRecentResource;
import org.pmiops.workbench.db.model.DbUserRecentlyModifiedResource;
import org.springframework.stereotype.Service;

@Service
public interface UserRecentResourceService {

  int USER_ENTRY_COUNT = 10;

  DbUserRecentResource updateNotebookEntry(
      long workspaceId, long userId, String notebookNameWithPath);

  void updateCohortEntry(long workspaceId, long userId, long cohortId);

  void updateConceptSetEntry(long workspaceId, long userId, long conceptSetId);

  void updateDataSetEntry(long workspaceId, long userId, long dataSetId);

  void updateCohortReviewEntry(long workspaceId, long userId, long cohortReviewId);

  void deleteNotebookEntry(long workspaceId, long userId, String notebookName);

  void deleteCohortEntry(long workspaceId, long userId, long cohortId);

  void deleteConceptSetEntry(long workspaceId, long userId, long conceptSetId);

  void deleteDataSetEntry(long workspaceId, long userId, long dataSetId);

  void deleteCohortReviewEntry(long workspaceId, long userId, long cohortReviewId);

  List<DbUserRecentResource> findAllResourcesByUser(long userId);

  List<DbUserRecentlyModifiedResource> findAllRecentlyModifiedResourcesByUser(long userId);
}

File: api/src/main/java/org/pmiops/workbench/db/dao/UserRecentResourceService.kt
package org.pmiops.workbench.db.dao

import java.sql.Timestamp
import org.pmiops.workbench.db.model.UserRecentResource
import org.springframework.stereotype.Service

@Service
interface UserRecentResourceService {

    fun updateNotebookEntry(
            workspaceId: Long, userId: Long, notebookNameWithPath: String, lastAccessDateTime: Timestamp): UserRecentResource

    fun updateCohortEntry(
            workspaceId: Long, userId: Long, cohortId: Long, lastAccessDateTime: Timestamp)

    fun updateConceptSetEntry(
            workspaceId: Long, userId: Long, conceptSetId: Long, lastAccessDateTime: Timestamp)

    fun deleteNotebookEntry(workspaceId: Long, userId: Long, notebookName: String)

    fun findAllResourcesByUser(userId: Long): List<UserRecentResource>
}
